exp/controllers/azuremachinepool_controller.go

```diff
@@ -329,7 +330,11 @@ func (ampr *AzureMachinePoolReconciler) reconcileNormal(ctx context.Context, mac
 			}
 
 			if reconcileError.IsTransient() {
-				log.Error(err, "failed to reconcile AzureMachinePool", "name", machinePoolScope.Name())
+				if azure.IsOperationNotDoneError(reconcileError) {
+					log.V(2).Info(fmt.Sprintf("AzureMachinePool reconcile not done: %s", reconcileError.Error()))
+				} else {
+					log.V(2).Info(fmt.Sprintf("transient failure to reconcile AzureMachinePool, retrying: %s", reconcileError.Error()))
+				}
```

Comment:
Modeled after the AzureCluster controller: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/95d3091c16926edc5ff5f4ad7b423bbb04f993de/controllers/azurecluster_controller.go#L229-L236